src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,545 @@
+characteristics, one for the LED and one for the temperature sensor.
+The former allows reading the current LED color as well as setting
+a new color. The latter allows subscribing to temperature change events.
+
+> NOTE: Initially I had some problems with the Bluetooth connection
+> being a bit unstable and not working on every startup, or bailing
+> out during connecting with a Frame Reassemble failure.
+>
+> If that happens, typing `rfkill block bluetooth` followed by
+> `rfkill unblock bluetooth` once over the serial connection, makes it
+> work again. The former issue went away when I started powering the
+> device from a power supply instead of just using USB for power.
+>
+> The Frame Reassemble failure happened when I sent temperature change
+> events too often.
+>
+> Generally you should always use external power when using Bluetooth
+> or connect something like a servo to your board.
+
+With noble, creating a characteristic is quite easy. All you need to do
+is to define how the characteristic communicates (options are read,
+write and notify, or any combination thereof) as well as define an uuid.
+Creating a new object and inheriting from the `bleno.Characteristic` is
+the easiest way.
+
+> NOTE: I am not using ES2016 here as Edison currently uses an older
+> version of Node.js.
+>
+> With the newly launched [Ostro Project](https://ostroproject.org)
+> which supports the Edison, that is no longer the case. Also, if you
+> use Brillo as part of the [Brillo](https://developers.google.com/brillo/) 
+> Early Access Program, then it is possible to compile and install a
+> recent version of Node.js.
+
+The resulting characteristic object looks like the following:
+
+{% highlight javascript %}
+var TemperatureCharacteristic = function() {
+  bleno.Characteristic.call(this, {
+    uuid: 'fc0a',
+    properties: ['read', 'notify'],
+    value: null
+  });
+    
+  this._lastValue = 0;
+  this._total = 0;
+  this._samples = 0;
+  this._onChange = null;
+};
+
+util.inherits(TemperatureCharacteristic, bleno.Characteristic);
+{% endhighlight %}
+
+We are storing the current temperature value in the `this._lastValue`
+variable. In order for a "read" to work we need to add an onReadRequest
+method and encode the value.
+
+{% highlight javascript %}
+TemperatureCharacteristic.prototype.onReadRequest = function(offset, callback) {
+  var data = new Buffer(8);
+  data.writeDoubleLE(this._lastValue, 0);
+  callback(this.RESULT_SUCCESS, data);
+};
+{% endhighlight %}
+
+For "notify" we need to add a method to handle subscription and
+unsubscription. Basically, we simply store a callback. When we
+have a new temperature reason we want to send, we then call that
+callback with the new value (encoded as above).
+
+{% highlight javascript %}
+TemperatureCharacteristic.prototype.onSubscribe = function(maxValueSize, updateValueCallback) {
+  console.log("Subscribed to temperature change.");
+  this._onChange = updateValueCallback;
```

Comment:
I was just thinking that using `_onChange` to store the callback, implied that you couldn't have several subscribers.